areaOfInterest/areas-of-interest.ts

```diff
@@ -0,0 +1,103 @@
+class AreasOfInterest {
+
+    preInit(config: any) {
+        this.config = config;
+    }
+
+    init(api: any) {
+        console.log(api);
+        //initialize attributes
+        let map = api; // map object that rv-extension loads into this file
+        let mapConfig = this.config;
+        let zones = mapConfig.map.components.areaOfInterest._source.areas;
```

Comment:
Is it possible to avoid the use of private properties here, `_source` should be avoided in an extension.

Reply:
Unfortunately this is not possible, seems to be a problem with how the config object is generated: 

![capture](https://user-images.githubusercontent.com/25359812/43909340-1807a81c-9bc8-11e8-9644-916f1539d520.PNG)

the previous plugin also accessed each area in the same way.

Reply:
update: this does not apply anymore because `areaOfInterest`  configs are how defined in a script tag